riocli/device/util.py

```diff
@@ -171,7 +177,84 @@ def is_remote_path(src, devices=[]):
     return None, src
 
 
-class DeviceNotFound(Exception):
-    def __init__(self, message='device not found'):
-        self.message = message
-        super().__init__(self.message)
+def create_hwil_device(spec, metadata):
+    """Create a new hardware-in-the-loop device."""
+    os = spec['virtual']['os']
+    codename = spec['virtual']['codename']
+    arch = spec['virtual']['arch']
+    labels = hwil_device_labels(spec.virtual.product, metadata.name)
+    device_name = f"{metadata['name']}-{spec['virtual']['product']}-{labels['user']}"
+    device_name = sanitize_hwil_device_name(device_name)
+    client = new_hwil_client()
+    try:
+        device_id = find_device_id(client, device_name)
+        return client.get_device(device_id)
+    except DeviceNotFound:
+        click.secho(
+            click.style(f'Creating hwil device with configuration = {os}:{codename}:{arch}', fg='cyan', bold=True))
+
+    try:
+        response = client.create_device(device_name, arch, os, codename, labels)
+        client.poll_till_device_ready(response.id, sleep_interval=5, retry_limit=3)
+        return response
+    except Exception as e:
+        click.secho(f'Failed to create hwil device: {str(e)}', fg='red')
+        raise e
+
+
+def execute_onboard_command(device_id: str, onboard_command: str) -> None:
+    """Execute the onboard command on a hardware-in-the-loop device."""
+    client = new_hwil_client()
+    try:
+        code, _, stderr = execute_command(client, device_id, onboard_command)
+        if code != 0:
+            raise Exception(f"Failed with exit code {code}: {stderr}")
+    except Exception as e:
+        raise e
+
+
+def hwil_device_labels(product_name, device_name) -> typing.Dict:
+    data = Configuration().data
+    user_email = data.get('email_id', None)
+    project_id = data.get('project_id', None)
+    organization_id = data.get('organization_id', None)
```

Comment:
todo: for this operation, these values will be there. so we need not `.get` them.
```suggestion
    user_email = data['email_id']
    project_id = data['project_id']
    organization_id = data['organization_id']
```